### PR TITLE
More impls in zerovec crate

### PR DESCRIPTION
**Pull request by sffc**
I keep hitting more impls that are missing. This PR adds even more.

CC @robertbastian because I added a Bakeable test in this PR to one type. It compares a string of code to actual code. If you like the style, I think we should add these to everywhere we have a Bakeable impl.

### CODE REVIEW

**Comment by Manishearth** [comment]
we can use insta for bake tests too, though I'd probably only use that for the bake crate itself

**Review by Manishearth — DISMISSED** [review]

**Comment by sffc** [comment]
I want to have the code actually get compiled. I don't know of a better way to do that than either a separate rs file (that gets compiled and also inspected as a string) or visual inspection with code followed by a string.

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
> I was just looking at this and then made #2063 because I don't like this test.

Thanks for taking https://github.com/unicode-org/icu4x/issues/2055 :smiley: 

I landed this because it blocked another PR (#2062). I'm happy if you want to fix up how I did the crabbake test.

File: utils/zerovec/src/flexzerovec/crabbake.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::{FlexZeroSlice, FlexZeroVec};
+use crabbake::*;
+
+impl Bakeable for FlexZeroVec<'_> {
+    fn bake(&self, env: &CrateEnv) -> TokenStream {
+        env.insert("core");
```
[the remark this answers is not in the document]
Comment (sffc) [reply utils/zerovec/src/flexzerovec/crabbake.rs:10]: I think I copied it from somewhere. Thanks for catching. I see you are fixing it in #2063.

**Pull request merged**